libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -220,48 +310,88 @@ class FeatureFlagConfigurationSettingsConverter implements ConfigurationSettings
+            if (filtersName) {
+              if (filtersName.toLowerCase() == "alwayson") {
+                featureFlagValue.conditions = defaultFeatureConditions;
+                break;
+              }
+              parameters.name = filtersName;
+              if (filtersParameters) {
+                parameters.parameters = filtersParameters;
+              }
+
+              featureFlagValue.conditions.clientFilters[i] = parameters as ClientFilter;
             }
-            parameters.name = filtersName;
-            if (filtersParameters) {
-              parameters.parameters = filtersParameters;
+            else {
+              throw new ArgumentError(
+                `This feature flag '${featureFlagName}' has a filter without the required 'name' property.`
+              );
             }
-
-            featureFlagValue.conditions.clientFilters[i] = parameters as ClientFilter;
-          }
-          else {
-            throw new ArgumentError(
-              `This feature flag '${featureFlagName}' has a filter without the required 'name' property.`
-            );
           }
         }
       }
+
+      const requirementType = featureData[requirementTypeKeyWord];
+
+      if (requirementType) {
+        if (this.validateRequirementType(featureFlagName, requirementType)) {
+          featureFlagValue.conditions.requirementType = requirementType;
+        }
+      }
     }
-    else if (typeof featureData[featureFlagName] == "boolean") {
-      featureFlagValue.enabled = featureData[featureFlagName];
+    else if (typeof featureData == "boolean") {
+      featureFlagValue.enabled = featureData;
     }
     else {
       throw new ArgumentError(
         `Data contains feature flags in invalid format. The type of ${featureFlagName} should be either boolean or dictionary.`
       );
     }
+
     return featureFlagValue;
   }
 
+  private getFeatureFlagDefinitionFromMsFmSchema(featureFlag: any): MsFeatureFlagValue {
+    const validate = this.ajv.compile<MsFeatureFlagValue>(featureFlagValueSchema);
+    const featureFlagCopy = JSON.parse(JSON.stringify(featureFlag)); //deep copy
+
+    // normalize client filters
+    if (featureFlagCopy.conditions && featureFlagCopy.conditions.client_filters) {
+      for (let i = 0; i < featureFlagCopy.conditions.client_filters.length; i++) {
+        featureFlagCopy.conditions.client_filters[i] = this.getLowerCaseFilters(featureFlagCopy.conditions.client_filters[i]);
+      }
+    }
+
+    const valid = validate(featureFlagCopy);
+    if (!valid) {
+      const validationError = new AjvValidationError(validate.errors as ErrorObject[]);
+      throw new ArgumentError(`Feature flag '${featureFlag.id}' is not in the correct format. ${validationError.getFriendlyMessage()}`);
+    }
+
+    const parsedFeatureFlag: MsFeatureFlagValue = {
+      ...featureFlag,
+      displayName: featureFlag.display_name,
+      conditions: {
+        clientFilters: featureFlag.conditions?.client_filters ?? []
```

Comment:
What about the `requirement_type`? Why not convert it as well?

Reply:
We should convert requirement_type as well. Updated